### PR TITLE
plugins.vkplay: new plugin

### PR DESCRIPTION
**Pull request by Inokenty90**
Plugin to support live streaming on vkplay.live

```
streamlink https://vkplay.live/sguhoman
[cli][info] Found matching plugin vkplay for URL https://vkplay.live/sguhoman
[cli][info] Available streams: 240p (worst), 360p, 480p, 720p, 1080p (best)
```

### CODE REVIEW

**Comment by bastimeyer** [comment]
Thanks for the PR. Adding a plugin for this site should be fine.

However, some code clean up needs to be done here first before this can be merged. Allow me to add some changes to your PR branch, so this doesn't get unnecessarily complicated. I will also add support for API error messages and for stream metadata.

**Comment by bastimeyer** [comment]
Metadata:
```
$ streamlink 'https://vkplay.live/d3l_by' best -j | jq .metadata
{
  "id": "4462052257156",
  "author": "d3l_by",
  "category": "Говорим и смотрим",
  "title": "ВЛОГ2 уже на ютубе!"
}
```

Offline:
```
$ streamlink 'https://vkplay.live/sguhoman'
[cli][info] Found matching plugin vkplay for URL https://vkplay.live/sguhoman
error: No playable streams found on this URL: https://vkplay.live/sguhoman
```

Non-existent:
```
$ streamlink 'https://vkplay.live/foo'
[cli][info] Found matching plugin vkplay for URL https://vkplay.live/foo
[plugins.vkplay][error] Blog not found
error: No playable streams found on this URL: https://vkplay.live/foo
```

File: src/streamlink/plugins/vkplay.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        self.category, self.title, streamdata = data
+        if not streamdata:
+            return
+
+        self.id, streams = streamdata
+
+        for streamtype, streamurl in streams:
+            if streamtype == "live_hls":
+                return HLSStream.parse_variant_playlist(self.session, streamurl)
```
Comment (mkbloke) [inline src/streamlink/plugins/vkplay.py:86]: Not that it seems to cause much of an issue in practice, but `streamurl` could be an empty string here.  I guess in practice that might not actually happen when `streamtype == "live_hls"` anyway, so I don't know if you'd consider it worth checking for.

**Review by mkbloke — APPROVED** [review]
Looks good to me.

File: src/streamlink/plugins/vkplay.py (hunk 1 above)
Comment (bastimeyer) [reply src/streamlink/plugins/vkplay.py:86]: I had a check for this before I commited my changes but then removed it. Let's add it back, though.

**Pull request merged**